server/src/main/java/org/elasticsearch/action/admin/cluster/coordination/CoordinationDiagnosticsAction.java

```diff
@@ -0,0 +1,153 @@
+import java.util.Objects;
+
+import static org.elasticsearch.cluster.coordination.CoordinationDiagnosticsService.CoordinationDiagnosticsResult;
+
+/**
+ * This action exposes CoordinationDiagnosticsService.diagnoseMasterStability() so that a node can get a remote node's view of
+ * coordination diagnostics (including master stability).
+ */
+public class CoordinationDiagnosticsAction extends ActionType<CoordinationDiagnosticsAction.Response> {
+
+    public static final CoordinationDiagnosticsAction INSTANCE = new CoordinationDiagnosticsAction();
+    public static final String NAME = "cluster:internal/coordination_diagnostics/info";
+
+    private CoordinationDiagnosticsAction() {
+        super(NAME, CoordinationDiagnosticsAction.Response::new);
+    }
+
+    public static class Request extends ActionRequest {
+        final boolean explain; // Non-private for testing
+
+        public Request(boolean explain) {
+            this.explain = explain;
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            return null;
+        }
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
+            this.explain = in.readBoolean();
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            out.writeBoolean(explain);
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            return explain == ((Request) o).explain;
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hashCode(explain);
+        }
+
+    }
+
+    public static class Response extends ActionResponse {
+
+        private final CoordinationDiagnosticsResult result;
+
+        public Response(StreamInput in) throws IOException {
+            super(in);
+            result = new CoordinationDiagnosticsResult(in);
+        }
+
+        public Response(CoordinationDiagnosticsResult result) {
+            this.result = result;
+        }
+
+        public CoordinationDiagnosticsResult getCoordinationDiagnosticsResult() {
+            return result;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            result.writeTo(out);
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            CoordinationDiagnosticsAction.Response response = (CoordinationDiagnosticsAction.Response) o;
+            return result.equals(response.result);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(result);
+        }
+    }
+
+    /**
+     * This transport action calls the CoordinationDiagnosticsService on a remote node.
```

Comment:
nit: This doc is rather misleading to me. The transport action exposes the functionality via the transport service (as opposed to calling a service on a remote node)